app/models/manageiq/providers/vmware/infra_manager/vim_connect_mixin.rb

```diff
@@ -53,6 +53,13 @@ def raw_connect(options)
       validate_connection do
         vim = MiqVim.new(ip, user, pass)
         raise MiqException::Error, _("Adding ESX/ESXi Hosts is not supported") unless vim.isVirtualCenter
+
+        # If the time on the vCenter is very far off from MIQ system time then
+        # any comparison of last_refresh_on and VMware TaskInfo.completeTime can be
+        # unreliable.
+        vc_time_diff = Time.parse(vim.currentTime).utc - Time.now.utc
+        raise MiqException::Error, _("vCenter time is too far out of sync with the ManageIQ system time") if vc_time_diff.abs > 10.minutes
```

Comment:
Sorry I just realized but I think this error would get back to the user, so we shouldn't have ManageIQ in there for production reasons.  I think we can just leave it out.

Reply:
:+1: done